compat/mingw.c

```diff
@@ -426,6 +426,67 @@ static wchar_t *normalize_ntpath(wchar_t *wbuf)
 	return wbuf;
 }
 
+static BOOL exists(const wchar_t *wpathname)
+{
+	HANDLE hnd = CreateFileW(wpathname, GENERIC_READ,
+				 FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE, NULL,
+				 OPEN_EXISTING, FILE_FLAG_BACKUP_SEMANTICS, NULL);
+	if (hnd == INVALID_HANDLE_VALUE)
+		return FALSE;
+	CloseHandle(hnd);
+	return TRUE;
+}
+
+static BOOL is_exclusively_writable(const wchar_t *wpathname)
+{
+	HANDLE hnd = CreateFileW(wpathname, GENERIC_WRITE, 0, NULL, OPEN_EXISTING, FILE_ATTRIBUTE_NORMAL, NULL);
+	if (hnd == INVALID_HANDLE_VALUE)
+		return !is_file_in_use_error(GetLastError());
```

Comment:
So, we only return false if we get a `file_in_use_error`. Do all other possible errors indicate that the file is exclusively writable?

Reply:
It can be just missing, which is fine for us. I didn't consider other cases. Do you have an example?

Reply:
The full list of possible error codes should be documented in CreateFileW. Did you check if `is_file_handel_in_use_error` covers all the error conditions that we are interested in?

Reply:
If the file is not writable for a different reason, renaming is unlikely to help, so we should just try and unlink it. I'll review other error codes anyway.

Reply:
According to https://stackoverflow.com/a/19249991/764870 there isn't any list of all possible errors. Anyway, the errors that I'm interested in are SHARING_VIOLATION and PERMISSION_DENIED, so `is_file_in_use_error` looks good.